web/src/buddy_list.js

```diff
@@ -86,6 +88,77 @@ export class BuddyList extends BuddyListConf {
+                placement = "auto";
+            }
+            const subscriber_count_function = this.total_subscriber_count;
+            tippy($elem[0], {
+                // Because the buddy list subheadings are potential click targets for purposes
+                // having nothing to do with the subscriber count (collapsing/expanding), we
+                // delay showing the tooltip until the user lingers a bit longer.
+                delay: [300, 0],
+                // Don't show tooltip on touch devices (99% mobile) since touch pressing on users in the left or right
+                // sidebar leads to narrow being changed and the sidebar is hidden. So, there is no user displayed
+                // to show tooltip for. It is safe to show the tooltip on long press but it not worth
+                // the inconvenience of having a tooltip hanging around on a small mobile screen if anything going wrong.
+                touch: false,
+                arrow: true,
+                placement,
+                showOnCreate: true,
+                onShow(instance) {
+                    let tooltip_text;
+                    const current_sub = narrow_state.stream_sub();
+                    const pm_ids_set = narrow_state.pm_ids_set();
+                    const subscriber_count = subscriber_count_function(current_sub, pm_ids_set);
+                    const elem_id = $elem.attr("id");
+                    if (elem_id === "buddy-list-users-matching-view-section-heading") {
+                        if (current_sub) {
+                            tooltip_text = $t(
+                                {
+                                    defaultMessage:
+                                        "{N, plural, one {# subscriber} other {# subscribers}}",
+                                },
+                                {N: subscriber_count},
+                            );
+                        } else {
+                            tooltip_text = $t(
+                                {
+                                    defaultMessage:
+                                        "{N, plural, one {# participant} other {# participants}}",
+                                },
+                                {N: subscriber_count},
+                            );
+                        }
+                    } else {
+                        const total_user_count = people.get_active_human_count();
+                        const other_users_count = total_user_count - subscriber_count;
+                        tooltip_text = $t(
+                            {
+                                defaultMessage:
+                                    "{N, plural, one {# other user} other {# other users}}",
+                            },
+                            {N: other_users_count},
+                        );
+                    }
+                    instance.setContent(tooltip_text);
```

Comment:
I opted for generating the text here instead of in the template because the wording is different depending on which of the three cases we're in. But it looks fairly ugly with the nested ifs -- is there a nicer way to do this?

Reply:
It seems like you ought to be able to make the assignment to `tooltip_text` just the once. The logic in any if/else statements seem like the should be limited to determining the appropriate `N` and the singular/plural noun forms. Maybe just constructing an object literal that gets interpolated into `defaultMessage` and `N`'s value? `{ N: <val>, singular: <val>, plural: <val> }? I don't know if that'd help, but maybe that gives you some alternate technique to stumble onto another one.

Reply:
Hm yeah I'm not sure if that would simplify it significantly, and I'm also not sure if that would make the translation strings harder to translate -- I think the words need to be shown in the context phrase that they're seen in. I think leaving it like this is best for now.

Reply:
I think this is probably fine as written, resolving. I'm pretty sure the translation library needs a direct `$t` call without indirection to find strings.